UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs

```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
 
                     baseNumber++;
                     var folderName = $"B{baseNumber}";
-                    checkFolder = FssBatchHelper.CheckforFolderExist(DownloadedFolderPath, folderName);
+                    checkFolder = FssBatchHelper.CheckforFolderExist(folderPath, folderName);
                 } while (checkFolder);
+
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenASerialEncFileIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenASerialEncFileIsGenerated()
         {
+            int mediaNumber = 1;
             int baseNumber = 1;
-            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            bool checkFolder;
+
+            foreach (string folderPath in DownloadedFolderPath)
             {
-                var FolderName = $"M0{mediaNumber}X02";
-                bool checkFolder;
-                DownloadedFolderPath = await FileContentHelper.ExchangeSetLargeFile(ApiEssResponse, FssJwtToken, FolderName);
                 do
                 {
-                    bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(DownloadedFolderPath, $"B{baseNumber}"), Config.ExchangeSetSerialEncFile);
-                    Assert.IsTrue(checkFile, $"{Config.ExchangeSetSerialEncFile} File not Exist in the specified folder path : {Path.Combine(DownloadedFolderPath, $"B{baseNumber}")}");
+                    bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(folderPath, $"B{baseNumber}"), Config.ExchangeSetSerialEncFile);
+                    Assert.IsTrue(checkFile, $"{Config.ExchangeSetSerialEncFile} File not Exist in the specified folder path : {Path.Combine(folderPath, $"B{baseNumber}")}");
 
                     //Verify Serial.ENC file content
-                    FileContentHelper.CheckSerialEncFileContentForLargeMediaExchangeSet(Path.Combine(DownloadedFolderPath, $"B{baseNumber}", Config.ExchangeSetSerialEncFile), baseNumber);
+                    FileContentHelper.CheckSerialEncFileContentForLargeMediaExchangeSet(Path.Combine(folderPath, $"B{baseNumber}", Config.ExchangeSetSerialEncFile), baseNumber);
 
                     baseNumber++;
                     var folderName = $"B{baseNumber}";
-                    checkFolder = FssBatchHelper.CheckforFolderExist(DownloadedFolderPath, folderName);
+                    checkFolder = FssBatchHelper.CheckforFolderExist(folderPath, folderName);
                 } while (checkFolder);
+
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenEncFilesAreGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenEncFilesAreGenerated()
         {
+            int mediaNumber = 1;
```

Comment:
I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

Reply:
Updated.

Thanks